etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/ruleengine/transformation/custom/MapLocalObservationCodes.java

```diff
@@ -0,0 +1,165 @@
+
+        var cwe =
+                HapiHelper.getCodingExtensionByUrl(coding, HapiHelper.EXTENSION_CWE_CODING)
+                        .getValue()
+                        .toString();
+        var codingSystem = HapiHelper.getCodingSystem(coding);
+
+        return Objects.equals(cwe, "alt-coding") && HapiHelper.LOCAL_CODE_URL.equals(codingSystem);
+    }
+
+    private void logUnmappedLocalCode(Bundle bundle, Coding coding) {
+        var msh41Identifier = HapiHelper.getMSH4_1Identifier(bundle);
+        var msh41Value = msh41Identifier != null ? msh41Identifier.getValue() : null;
+
+        logger.logWarning(
+                "Unmapped local code detected: '{}', from sender: '{}', message Id: '{}'",
+                coding.getCode(),
+                msh41Value,
+                HapiHelper.getMessageControlId(bundle));
+    }
+
+    private Coding getMappedCoding(IdentifierCode identifierCode) {
+        var mappedCoding =
+                new Coding(
+                        HapiHelper.urlForCodeType(identifierCode.codingSystem()),
+                        identifierCode.code(),
+                        identifierCode.display());
+        mappedCoding.addExtension(HapiHelper.EXTENSION_CWE_CODING, new StringType("coding"));
+
+        mappedCoding.addExtension(
+                HapiHelper.EXTENSION_CODING_SYSTEM, new StringType(identifierCode.codingSystem()));
+
+        return mappedCoding;
+    }
+
+    /**
+     * Initializes the local-to-LOINC/PLT hash map, customized for CDPH and UCSD. Currently, the
+     * mapping is hardcoded for simplicity. If expanded to support additional entities, the
+     * implementation may be updated to allow dynamic configuration via
+     * transformation_definitions.json or a database-driven mapping.
+     */
+    private void initMap() {
+        this.codingMap = new HashMap<>();
+        codingMap.put(
+                "99717-32",
+                new IdentifierCode(
+                        "85269-9",
+                        "X-linked Adrenoleukodystrophy (X- ALD) newborn screen interpretation",
+                        HapiHelper.LOINC_CODE));
+        codingMap.put(
+                "99717-33",
+                new IdentifierCode(
+                        "85268-1",
+                        "X-linked Adrenoleukodystrophy (X- ALD) newborn screening comment-discussion",
+                        HapiHelper.LOINC_CODE));
+        codingMap.put(
+                "99717-34",
+                new IdentifierCode(
+                        "PLT325",
+                        "ABCD1 gene mutation found [Identifier] in DBS by Sequencing",
+                        HapiHelper.PLT_CODE));
+        codingMap.put(
+                "99717-6",
+                new IdentifierCode(
+                        "53340-6",
+                        "17-Hydroxyprogesterone [Moles/volume] in DBS",
+                        HapiHelper.LOINC_CODE));
+        // map.put("99717-35", new IdentifierCode("REQUEST_PLT", "REQUEST_PLT",
```

Comment:
Was the commented mappings left on purpose? If so, it'd be helpful to have a comment explaining why. Otherwise is better to remove

Reply:
We've received the final mappings and updated the hash map, so no more commented lines. 😄